### PR TITLE
Resolving issue #1920

### PR DESCRIPTION
**Pull request by netniV**
Added two new functions `cacti_sizeof()` and `cacti_count()` which will always return a numeric value.  If the variable passed is not an array or is false, a zero will be returned.  Otherwise, the base PHP count/sizeof function result is returned.

### CODE REVIEW

File: lib/functions.php [hunk 1]
```diff
@@ -4853,3 +4853,11 @@ function cacti_ptoa($title, $addr) {
 		print str_pad(dechex(ord($char)),2,'0',STR_PAD_LEFT);
 	}
 }
+
+function cacti_sizeof($array) {
+	return ($array === false || !is_array($array)) ? 0 : cacti_sizeof($array);
+}
+
+function cacti_count($array) {
+	return ($array === false || !is_array($array)) ? 0 : cacti_count($array);
```
Comment (pautiina) [inline lib/functions.php]: maybe need:
`return ($array === false || !is_array($array)) ? 0 : count($array);`

File: lib/functions.php [hunk 2]
```diff
@@ -4853,3 +4853,11 @@ function cacti_ptoa($title, $addr) {
 		print str_pad(dechex(ord($char)),2,'0',STR_PAD_LEFT);
 	}
 }
+
+function cacti_sizeof($array) {
+	return ($array === false || !is_array($array)) ? 0 : cacti_sizeof($array);
```
Comment (pautiina) [inline lib/functions.php]: maybe need:
`return ($array === false || !is_array($array)) ? 0 : sizeof($array);`

**Comment by cigamit** [comment]
You should have asked before doing this.  I'm not a fan of these wrappers.

**Comment by cigamit** [comment]
Just read the issue, 7.2 is bleeding edge.  I would prefer to wait till this is fixed in 7.2.

File: lib/functions.php (hunk 2 above)
Comment (netniV) [reply lib/functions.php]: Good spot, I meant to revert that last one when I did my global replace.

**Comment by netniV** [comment]
Also, ubuntu and Arch are two linux distributions definitely shipping with 7.2 by default.

**Comment by cigamit** [comment]
Turns out that this is a behavior change in PHP 7.2 not a bug.  So, we are pretty much stuck with the 'easy' fix in the mean time.  Don't have to use it everywhere, but might as well.

**Comment by netniV** [comment]
Yes, as mentioned elsewhere, plugins that are changed to use this, will be forced to a min v1.2 requirement.

**Pull request merged**